lib/galaxy/managers/configuration.py

```diff
@@ -29,67 +29,67 @@ def default_serializer(self, config, key):
-            'ga_code'                           : _defaults_to(None),
-            'enable_unique_workflow_defaults'   : _defaults_to(False),
+            'enable_quotas'                     : _defaults_to(self.app.config.enable_quotas),
+            'remote_user_logout_href'           : _defaults_to(self.app.config.remote_user_logout_href or ''),
+            'datatypes_disable_auto'            : _defaults_to(self.app.config.datatypes_disable_auto),
+            'allow_user_dataset_purge'          : _defaults_to(False),  # schema default is True
+            'ga_code'                           : _defaults_to(self.app.config.ga_code),
+            'enable_unique_workflow_defaults'   : _defaults_to(self.app.config.enable_unique_workflow_defaults),
             'has_user_tool_filters'             : _defaults_to(False),
 
             # TODO: is there no 'correct' way to get an api url? controller='api', action='tools' is a hack
             # at any rate: the following works with path_prefix but is still brittle
             # TODO: change this to (more generic) upload_path and incorporate config.nginx_upload_path into building it
-            'nginx_upload_path'                 : lambda i, k, **c: getattr(i, k, False),
-            'chunk_upload_size'                 : _defaults_to(104857600),
-            'ftp_upload_site'                   : _defaults_to(None),
+            'nginx_upload_path'                 : lambda item, key, **context: getattr(item, key, False),
+            'chunk_upload_size'                 : _defaults_to(self.app.config.chunk_upload_size),
+            'ftp_upload_site'                   : _defaults_to(self.app.config.ftp_upload_site),
             'version_major'                     : _defaults_to(None),
-            'require_login'                     : _defaults_to(None),
-            'inactivity_box_content'            : _defaults_to(None),
-            'visualizations_visible'            : _defaults_to(True),
-            'interactivetools_enable'           : _defaults_to(False),
-            'message_box_content'               : _defaults_to(None),
-            'message_box_visible'               : _defaults_to(False),
-            'message_box_class'                 : _defaults_to('info'),
-            'server_startttime'                 : lambda i, k, **c: server_starttime,
-            'mailing_join_addr'                 : _defaults_to('galaxy-announce-join@bx.psu.edu'),
-            'server_mail_configured'            : lambda i, k, **c: True if getattr(i, 'smtp_server', False) else False,
-            'registration_warning_message'      : _defaults_to(None),
-            'welcome_url'                       : _defaults_to(None),
-            'show_welcome_with_login'           : _defaults_to(True),
-            'cookie_domain'                     : _defaults_to(None),
+            'require_login'                     : _defaults_to(self.app.config.require_login),
```

Comment:
This looks really confused right - since ``_defaults_to`` is what is used if ``config`` doesn't have the attribute but the default your filling in is an unprotected access of that exact value from ``config``. This would all read a lot better I think if ``_defaults_to(self.app.config.require_login)`` was just ``_required_attribute``

with 

```python
            def _required_attribute(i, k, **c):
                assert hasattr(i, k)
                return getattr(i, k)
```

Reply:
I’m not suggesting you should use a protected access pattern here.

Reply:
```python
'require_login': _defaults_to(self.app.config.require_login),
```

causes _defaults_to to be applied to config right? So you're effectively saying:

```python
getattr(self.app.config, 'require_login', self.app.config.require_login)
```

better just to annotate that it has a required value in some fashion. My idea was:

```python
'require_login': _required_value,
```

Reply:
Ah, of course! Yes, this makes sense. I'll make the change. Thanks!